File: backend/app/models/session.py
from datetime import datetime
from enum import Enum

from pydantic import BaseModel


class SessionStatus(str, Enum):
    STARTED = "started"
    COMPLETED = "completed"


class PhaseQuality(str, Enum):
    OK = "ok"
    PARTIAL = "partial"
    MISSED = "missed"


class PhaseResult(BaseModel):
    phase_index: int
    phase_name: str
    participation: bool
    hold_achieved: bool | None = None
    elbow_participation: bool | None = None
    quality: PhaseQuality = PhaseQuality.MISSED
    notes: str = ""


class SessionCreateRequest(BaseModel):
    flow_id: str


class SessionCreateResponse(BaseModel):
    session_id: str
    flow_id: str
    status: SessionStatus
    created_at: datetime


class SessionResultRequest(BaseModel):
    phase_results: list[PhaseResult]
    duration_actual_sec: float


class Session(BaseModel):
    session_id: str
    flow_id: str
    user_id: str
    status: SessionStatus
    created_at: datetime
    completed_at: datetime | None = None
    phase_results: list[PhaseResult] = []
    duration_actual_sec: float | None = None
    summary: str | None = None

File: frontend/types/session.ts
export type PhaseQuality = "ok" | "partial" | "missed";

export interface PhaseResult {
  phase_index: number;
  phase_name: string;
  participation: boolean;
  hold_achieved: boolean | null;
  elbow_participation: boolean | null;
  quality: PhaseQuality;
  notes: string;
}

export interface SessionCreateResponse {
  session_id: string;
  flow_id: string;
  status: string;
  created_at: string;
}

export interface SessionResultRequest {
  phase_results: PhaseResult[];
  duration_actual_sec: number;
}

export interface Session {
  session_id: string;
  flow_id: string;
  user_id: string;
  status: string;
  created_at: string;
  completed_at: string | null;
  phase_results: PhaseResult[];
  duration_actual_sec: number | null;
  summary: string | null;
}
